message.go

```diff
@@ -43,7 +47,29 @@ type Message struct {
 	UpdatedAt *time.Time `json:"updated_at,omitempty"`
 
 	// any other fields the user wants to attach a message
-	ExtraData map[string]interface{}
+	ExtraData map[string]interface{} `json:"-"`
```

Comment:
How about having a custom type `ExtraData` that implements these interfaces and other types embeds it. Can we prevent duplication ?

Reply:
No, I tried this but it gets all wonky, it is also a backwards incompatible change.

Reply:
> backwards incompatible change

Yes.

With a version bump should be fine though I believe. Maybe with v3 series.